src/index.ts

```diff
@@ -306,13 +299,16 @@ export class CruxClient {
                     throw errors.ErrorHelper.getPackageError(null, errors.PackageErrorCode.ExistingCruxIDFound, this._payIDClaim.virtualAddress);
                 }
 
-                let identityClaim: nameService.IIdentityClaim;
+                let identityClaim: nameService.IIdentityClaim | undefined;
+                await this._decryptKeyPair();
                 identityClaim = {secrets: {identityKeyPair: this._keyPair}};
```

Comment:
what's the difference b/w identityClaim and this._keyPair. As it creating two variables with the private key in memory.

Reply:
there is just a difference of object interface.